classes/core/PKPHtmlSanitizer.php

```diff
@@ -0,0 +1,159 @@
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPHtmlSanitizer
+ *
+ * @ingroup core
+ *
+ * @brief Wrapper on the top of Symfony's HtmlSanitizer implementation
+ *
+ */
+
+namespace PKP\core;
+
+use Illuminate\Support\Str;
+use Illuminate\Support\Collection;
+use PKP\config\Config;
+use Symfony\Component\HtmlSanitizer\HtmlSanitizer;
+use Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig;
+use Symfony\Component\HtmlSanitizer\Reference\W3CReference;
+
+class PKPHtmlSanitizer
+{
+    /**
+     * Collection of valid element specify by W3C Sanitizer API
+     * 
+     * @see const HEAD_ELEMENTS and BODY_ELEMENTS at \Symfony\Component\HtmlSanitizer\Reference\W3CReference
+     * @see https://wicg.github.io/sanitizer-api/#default-configuration
+     */
+    protected Collection $w3cValidElements;
+
+    /**
+     * Instance of HtmlSanitizerConfig config
+     */
+    protected HtmlSanitizerConfig $htmlSanitizerConfig;
+
+    /**
+     * Create a new instance
+     */
+    public function __construct(string $allowable)
+    {
+        $this->w3cValidElements = collect(
+            array_merge(W3CReference::HEAD_ELEMENTS, W3CReference::BODY_ELEMENTS)
+        )->keys();
+
+        $this->buildSanitizerConfig(
+            $this->generateAllowedTagToAttributeMap(
+                Config::getVar('security', $allowable, null) ?? $allowable
+            )
+        );
+    }
+
+    /**
+     * Set a pre defined instance of \Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig
+     */
+    public function setSanitizerConfig(HtmlSanitizerConfig $htmlSanitizerConfig): self
+    {
+        $this->htmlSanitizerConfig = $htmlSanitizerConfig;
+
+        return $this;
+    }
+
+    /**
+     * Get the instance of \Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig
+     */
+    public function getSanitizerConfig(): HtmlSanitizerConfig
+    {
+        return $this->htmlSanitizerConfig;
+    }
+
+    /**
+     * Sanitize the given html string
+     */
+    public function sanitize(string $html): string
+    {   
+        return (new HtmlSanitizer($this->htmlSanitizerConfig))->sanitize(
+            // Here we are removing any html tags that should not be handled by sanitizer
+            strip_tags($html, $this->getSanitizableTags()->toArray())
+        );
+    }
+
+    /**
+     * Build up the \Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig instance
+     */
+    protected function buildSanitizerConfig(Collection $allowedTagToAttributeMap): void
+    {
+        $this->htmlSanitizerConfig = (new HtmlSanitizerConfig())
+            ->allowLinkSchemes(['https', 'http', 'mailto'])
+            ->allowMediaSchemes(['https', 'http']);
+        
+        if ($allowedTagToAttributeMap->count()) {
+            $allowedTagToAttributeMap->each(
+                fn (array $attributes, string $tag) => $this->htmlSanitizerConfig = $this->htmlSanitizerConfig->allowElement($tag, $attributes)
```

Comment:
Shouldn't `$this->htmlSanitizerConfig` be appended to rather than reinitialized on multiple collection elements?

Reply:
the config initialized right before the looping of `allowedTagToAttributeMap` as
```php
$this->htmlSanitizerConfig = (new HtmlSanitizerConfig())
            ->allowLinkSchemes(['https', 'http', 'mailto'])
            ->allowMediaSchemes(['https', 'http']);
```

Reply:
Ah, I was getting confused by the habit of cloning of the config object within the Symfony toolset! I see why they did it that way, but it certainly makes this part harder to read.